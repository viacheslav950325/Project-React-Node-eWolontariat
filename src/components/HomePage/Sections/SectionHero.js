import { Box, Typography } from "@material-ui/core";
import { Button } from "@material-ui/core";
import LogoPomocny from "../../../assets/img/logo-pomocny.svg";
import HeroImg from "../../../assets/img/hero-img.svg";
import { styled } from '@mui/material/styles';

const StyledBox = styled(Box)(({ theme }) => ({
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    padding: "2rem",
    margin: "2rem 0",
    height: "100%",
    justifyContent: "center",
    alignItems: "center",
    [theme.breakpoints.down('md')]: {
        flexDirection: "column",
        padding: "0", 
        '& p': {
            fontSize: "1rem",
            margin: "2rem 0.8rem",
            paddingRight: "0",
        },
        '& .box-responsive': {
            display: "block",
            padding: "1.4rem",
            textAlign: "center"
        },
        '& img': {
            maxWidth: "70%",
            paddingLeft: "0",
        },
        '& .main-img': {
            padding: "1rem",
            margin: "1rem"
        }
    },
  }));

const SectionHero = () => {

    return(
        <StyledBox id={"section-hero"}>
            <Box className={"box-responsive"} display={'flex'} padding={'3rem'} justifyContent={'space-evenly'}>
                <Box alignSelf={'center'}>
                    <Box className="logo" component="img" height={'80px'} alt="Logo pomocny.pl" src={LogoPomocny}/>
                    <Typography variant='body1' paragraph gutterBottom={true}>
                        Miejsce spotkań dla tych, którzy potrzebują <br/> pomocy i dla tych, którzy tej pomocy chętnie udzielą.
                    </Typography>
                    <Button 
                        variant="contained" 
                        color='secondary'
                        size={'large'}  
                        href="#section-how-it-works">Zobacz jak to działa</Button>
                </Box>
                <Box className={"main-img"} style={{paddingLeft: "2rem"}} component="img" maxWidth={'400px'} alt="" src={HeroImg}/>
            </Box>
        </StyledBox>
    )
}

export default SectionHero;