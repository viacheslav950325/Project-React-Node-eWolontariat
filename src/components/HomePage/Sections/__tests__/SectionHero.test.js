import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SectionHero from "../SectionHero";

describe("SectionHero",()=>{
    it("Should render SectionHero correctly",()=>{
        const wrapper = shallow(<SectionHero/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});