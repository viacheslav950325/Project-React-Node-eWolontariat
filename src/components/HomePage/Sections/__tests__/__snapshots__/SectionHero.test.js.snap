// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHero Should render SectionHero correctly 1`] = `
<Styled(Styled(MuiBox))
  id="section-hero"
>
  <Styled(MuiBox)
    className="box-responsive"
    display="flex"
    justifyContent="space-evenly"
    padding="3rem"
  >
    <Styled(MuiBox)
      alignSelf="center"
    >
      <Styled(MuiBox)
        alt="Logo pomocny.pl"
        className="logo"
        component="img"
        height="80px"
        src="logo-pomocny.svg"
      />
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        paragraph={true}
        variant="body1"
      >
        Miejsce spotkań dla tych, którzy potrzebują 
        <br />
         pomocy i dla tych, którzy tej pomocy chętnie udzielą.
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        href="#section-how-it-works"
        size="large"
        variant="contained"
      >
        Zobacz jak to działa
      </WithStyles(ForwardRef(Button))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      alt=""
      className="main-img"
      component="img"
      maxWidth="400px"
      src="hero-img.svg"
      style={
        Object {
          "paddingLeft": "2rem",
        }
      }
    />
  </Styled(MuiBox)>
</Styled(Styled(MuiBox))>
`;
